platform/ios/src/MGLMapView.mm

```diff
@@ -2727,6 +2727,11 @@ - (void)removeStyleClass:(NSString *)styleClass
     {
         return pair.second.annotation;
     });
+    
+    annotations.erase(std::remove_if(annotations.begin(), annotations.end(),
+                                     [](const id <MGLAnnotation> annotation) { return annotation == nullptr; }),
+                      annotations.end());
+    
     return [NSArray arrayWithObjects:&annotations[0] count:annotations.size()];
```

Comment:
That map should never contain the user location annotation, since it never gets a tag from mbgl.

Can you assert instead of removing the annotation in this case? I think you've uncovered a case where we're using [subscript] notation on a std::map without first checking for existence – that creates a bogus entry in the map, which is not what you'd expect.